src/hooks/useUserProfiles.tsx

```diff
@@ -0,0 +1,151 @@
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import { useAppConfig } from './useAppConfig';
+import { useActiveAccount } from './useActiveAccount';
+import { useHttpClient } from './useHttpClient';
+import { compact, uniq } from 'lodash';
+import { useCache } from './CacheProvider';
+import { useUser } from './useUser';
+import { useAuth } from './useAuth';
+
+export type UserProfile = {
+  id: string;
+  profile: { displayName?: string; picture?: string; givenName?: string };
+};
+
+type UserProfilesContextProps = {
+  getProfiles: () => UserProfile[];
+  isLoading?: boolean;
+};
+
+export const UserProfilesContext = createContext<UserProfilesContextProps>({
+  getProfiles: () => [],
+  isLoading: undefined,
+});
+
+export const UserProfilesContextProvider = ({
+  children,
+}: {
+  children?: React.ReactNode;
+}) => {
+  const { accountHeaders, isLoading, account } = useActiveAccount();
+  const { apiClient } = useHttpClient();
+  const { cache } = useCache();
+  const appConfig = useAppConfig();
+  const profiles = useRef<Map<string, UserProfile>>(new Map());
+  const { isLoggedIn } = useAuth();
+  const placeHolderProfile = (userId: string) => ({
+    id: userId,
+    profile: {
+      displayName: userId,
+    },
+  });
+  const messageTiles = appConfig.data?.homeTab?.messageTiles;
+  const userIds = uniq(messageTiles?.flatMap((tile) => tile.userIds));
```

Comment:
This is named `useUserProfiles`, but it looks coupled to messageTiles.  Let's either rename to be more specific, or refactor to be more re-usable (e.g. by taking in `userIds` as a prop).

Reply:
Yeah, good point.

Reply:
Refactored and renamed